WIP: StateChangeFinder cannot be made a singleton

IceAndSnow opened the pull request:
Closes #134

IceAndSnow committed:
Remove state from RapidCore.Diffing.StateChangeFinder

pingvinen commented:
@IceAndSnow how come there are all these code changes, when it could (based on the issue) just be a matter of moving `worker = new StateChangeFinderWorker(new InstanceTraverser());` into `GetChanges(..)`?

What am I missing?

IceAndSnow commented:
@pingvinen It is in order to make it more testable. If your suggested change was used, the worker could not be faked in a unit-test.

pingvinen committed:
Remove knowledge of InstanceTraverser from StateChangeFinder

The InstanceTraverser can still be injected, as the caller can simply
use the workerGenerator to inject it.

pingvinen reviewed (approved):
With the changes I pushed, I think this change becomes ok.

As far as I can see, these changes do not require any changes to the documentation.

IceAndSnow commented:
I think it looks good :+1:
